include/hc_printf.hpp

```diff
@@ -23,23 +23,35 @@ union PrintfPacketData {
   float           f;
   void*           ptr;
   const void*     cptr;
-  std::atomic_int ai;
+
+  // Members al, uia, ull used by headers for buffer offset insts.
+  // There are two offsets, the PrintfPacket buffer offset uia[0] and
+  // the Printf string buffer offset uia[1]. We want to update a
+  // single atomic offset al, so that both buffers will increment
+  // in order with respect to each other. This way, offsets can be
+  // managed by single ullong 8B, which is divided into 2 4B uints.
```

Comment:
@aaronenyeshi description here doesn't seem to clearly explain the relationship between `al`, `uia[]`, and `ull`. there is no explaination about what `ull` would be used.